forte/mrdsrg-spin-adapted/sa_mrdsrg.cc

```diff
@@ -84,6 +84,13 @@ void SA_MRDSRG::startup() {
     // link F_ with Fock_ of SADSRG
     F_ = Fock_;
 
+    // bypass the check when doing ASET (B)
+    bool semi_skip = foptions_->get_bool("EMBEDDING_DISABLE_SEMI_CHECK");
+    bool embedding = foptions_->get_bool("EMBEDDING");
+    if (embedding && semi_skip) {
```

Comment:
Is it still necessary after the refactorization? It seems the test cases do not enter here.

Reply:
I believe it is still needed. Since for A + B computation we need to turn off semi check, and for A computation we need to follow the option "semicanonical".
I do not have a test case where the environment (B) use sa-mrdsrg. I have it here just for completeness.

Reply:
> I believe it is still needed. Since for A + B computation we need to turn off semi check, and for A computation we need to follow the option "semicanonical".

Why do we need to turn off this check for A+B?

Reply:
Because we can only semi-canonicalize the Fock matrix in one way, not both, so either A or A+B computation will be not-semicanonicalized. However, we cannot let the semicanonicalization codes kick in. The figure 2 in the paper shows the situation.

Reply:
I see, that makes sense. Thanks for the clarification.

Reply:
Thanks for your comment !~